libs/dh/message-archive/feature-log-search/src/lib/searchresult/dh-message-archive-log-search-result.component.ts

```diff
@@ -16,66 +16,152 @@
+import { MatTableDataSource, MatTableModule } from '@angular/material/table';
+import { MatSort, MatSortModule } from '@angular/material/sort';
+import { TranslocoModule, TranslocoService } from '@ngneat/transloco';
 import { DhSharedUiDateTimeModule } from '@energinet-datahub/dh/shared/ui-date-time';
 import { MessageArchiveSearchResultItemDto } from '@energinet-datahub/dh/shared/domain';
 import { WattIconModule } from '@energinet-datahub/watt/icon';
 import { WattButtonModule } from '@energinet-datahub/watt/button';
 import { WattBadgeModule } from '@energinet-datahub/watt/badge';
+import { WattSpinnerModule } from '@energinet-datahub/watt/spinner';
 import { WattEmptyStateModule } from '@energinet-datahub/watt/empty-state';
+import {
+  WattTableComponent,
+  WattTableColumnDef,
+  WattTableCellDirective,
+} from '@energinet-datahub/watt/table';
+import { WattCardModule } from '@energinet-datahub/watt/card';
+import { ToLowerSort } from '@energinet-datahub/dh/shared/util-table';
+import { DocumentTypes } from '@energinet-datahub/dh/message-archive/domain';
+import { WattDropdownOptions } from '@energinet-datahub/watt/dropdown';
+
+import {
+  DhMessageArchiveDrawerComponent,
+  DhMessageArchiveDrawerScam,
+} from '../drawer/dh-message-archive-drawer.component';
+
+import { ActorNamePipe } from '../shared/dh-message-archive-actor.pipe';
+import { DocumentTypeNamePipe } from '../shared/dh-message-archive-documentTypeName.pipe';
+
+import { DhMessageArchiveStatusComponentScam } from '../shared/dh-message-archive-status.component';
 
 @Component({
   changeDetection: ChangeDetectionStrategy.OnPush,
   selector: 'dh-message-archive-log-search-result',
   templateUrl: './dh-message-archive-log-search-result.component.html',
   styleUrls: ['./dh-message-archive-log-search-result.component.scss'],
 })
-export class DhMessageArchiveLogSearchResultComponent {
-  @Input() searchResult: Array<MessageArchiveSearchResultItemDto> | null = [];
+export class DhMessageArchiveLogSearchResultComponent
+  implements AfterViewInit, OnChanges
+{
+  private _translateScrope = 'messageArchive.search';
+  private _selectedRow?: string;
+  @ViewChild(MatSort) matSort!: MatSort;
+  @ViewChild(DhMessageArchiveDrawerComponent)
+  messageDrawer!: DhMessageArchiveDrawerComponent;
+  @Input() searchResult: Array<MessageArchiveSearchResultItemDto> = [];
   @Output() showLogDownloadPage =
     new EventEmitter<MessageArchiveSearchResultItemDto>();
   @Output() downloadLogFile =
     new EventEmitter<MessageArchiveSearchResultItemDto>();
-  @Input() isSearching: boolean | null = false;
-  displayedColumns: string[] = [
-    'messageId',
-    'rsmName',
-    'sender',
-    'createddate',
-    'status',
-    'logoptions',
-  ];
+  @Input() isSearching = false;
+
+  columns: WattTableColumnDef<MessageArchiveSearchResultItemDto>;
+
+  @Input() hasSearchError = false;
+  @Input() isInit = false;
+  @Input() actors!: WattDropdownOptions;
+
+  DocumentTypes = DocumentTypes;
```

Comment:
suggestion: this doesn't seem to be used in the template anymore. Can be removed.

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1178/commits/1969319bb5400e977b8671225165b5eef0c1e381